### PR TITLE
Upgrade dependencies for asset-publisher

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: dockers/asset-publisher/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-FROM golang:1.14.6-alpine3.12 AS builder
+FROM golang:1.17.3-alpine3.15 as builder
 WORKDIR /tool
 COPY main.go .
 COPY go.mod .
 COPY go.sum .
```
Comment (knanao) [inline dockers/asset-publisher/Dockerfile:5]: [nits] It;s seems better to reduce the number of layers ASAP.
```suggestion
COPY main.go go.mod go.sum .
```

**Comment by nghialv** [comment]
Let me merge this. 😄 
/approve

File: dockers/asset-publisher/Dockerfile (hunk 1 above)
Comment (knanao) [reply dockers/asset-publisher/Dockerfile:5]: oh, it's too late👼

Comment (nghialv) [reply dockers/asset-publisher/Dockerfile:5]: Oops. Nice catch! Let me fix them in another PR.

**Pull request merged**